Feature/mfs 25 criar um metodo que retorna dados estatisticos de consumo e

AndreLFernandesDev opened the pull request.

AndreLFernandesDev committed:
feat:
Creating a method that returns statistical data on consumption and supply expenses

AndreLFernandesDev committed:
feat:
Creating a method that returns statistical data on consumption and supply expenses

AndreLFernandesDev committed:
feat:
Creating a method that returns statistical data on consumption and supply expenses

mayconfsousa commented on VehiTrack/Services/RefuelingRecordService.cs:
```diff
@@ -126,10 +125,254 @@ int vehicleId
+            int idVehicle
+        )
+        {
+            var refuelingRecordsExtend = await GetExtendedRefuelingRecordsByVehicleId(idVehicle);
```
Esse método `GetStatisticalDataOnConsumptionAndSupplyCosts` está retornando 118 items ao invés de 119, para vehicleId = 1. Isso está afetando o resultado dos outros.

mayconfsousa commented on VehiTrack/Services/RefuelingRecordService.cs:
```diff
@@ -126,10 +125,254 @@ int vehicleId
+                .Select(r => new
+                {
+                    Type = r.Key,
+                    TotalLyters = Math.Round(r.Sum(r => r.Quantity), 2)
```
```suggestion
                    TotalLiters = Math.Round(r.Sum(r => r.Quantity), 2)
```

mayconfsousa commented on VehiTrack/Services/RefuelingRecordService.cs:
```diff
@@ -126,10 +125,254 @@ int vehicleId
+                .Select(g => new
+                {
+                    Type = g.Key,
+                    Average = Math.Round(g.Sum(r => r.Consumption) / g.Count(), 2)
```
Usar `g.Average`

mayconfsousa commented on VehiTrack/Services/RefuelingRecordService.cs:
```diff
@@ -126,10 +125,254 @@ int vehicleId
+                });
+
+            //Average consumption per month/year (by type of fuel)
+            var averageCOnsumptionPerMonthYear = refuelingRecordsExtend
```
```suggestion
            var averageConsumptionPerMonthYear = refuelingRecordsExtend
```

mayconfsousa commented on VehiTrack/Services/RefuelingRecordService.cs:
```diff
@@ -126,10 +125,254 @@ int vehicleId
+                });
+
+            //Better consumption (by type of fuel)
+            var betterConsumption = refuelingRecordsExtend
```
```suggestion
            var bestConsumption = refuelingRecordsExtend
```

AndreLFernandesDev committed:
feat:
Creating a method that returns statistical data on consumption and supply expenses

mayconfsousa commented on VehiTrack/Services/RefuelingRecordService.cs:
```diff
@@ -1,3 +1,4 @@
+using System.Security.Principal;
```
Acho que pode remover esse `using` se não tiver sendo usado.

AndreLFernandesDev committed:
feat:
Creating a method that returns statistical data on consumption and supply expenses
